src/Controller/Admin/AdditionalContactDataController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\Contact;
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use HandcraftedInTheAlps\RestRoutingBundle\Controller\Annotations\RouteResource;
+use HandcraftedInTheAlps\RestRoutingBundle\Routing\ClassResourceInterface;
+use Sulu\Bundle\ContactBundle\Admin\ContactAdmin;
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Security\SecuredControllerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * @RouteResource("additional-contact-data")
+ */
+class AdditionalContactDataController extends AbstractRestController implements ClassResourceInterface, SecuredControllerInterface
+{
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ViewHandlerInterface $viewHandler,
+        ?TokenStorageInterface $tokenStorage = null
+    ) {
+        $this->entityManager = $entityManager;
+
+        parent::__construct($viewHandler, $tokenStorage);
+    }
+
+    public function getAction(int $id): Response
+    {
+        $contact = $this->entityManager->getRepository(Contact::class)->find($id);
```

Comment:
its recommended to reference by the `ContactInterface` if its a bundle and somebody would again override the entity it still works in that case.

Reply:
Not sure if this is a case of the interface but to avoid PHPStan errors:

```php
        /** @var Contact|null */
        $contact = $this->entityManager->getRepository(Contact::class)->find($id);
```